superset-frontend/src/SqlLab/components/SqlEditor/index.tsx

```diff
@@ -0,0 +1,742 @@
+      sqlEditorId: qe.id,
+      tab: qe.title,
+      schema: qe.schema,
+      tempTable: ctasArg ? ctas : '',
+      templateParams: qe.templateParams,
+      queryLimit: qe.queryLimit || defaultQueryLimit,
+      runAsync: database ? database.allow_run_async : false,
+      ctasArg,
+      ctas_method,
+      updateTabState: !qe.selectedText,
+    };
+    dispatch(runQuery(query));
+    dispatch(setActiveSouthPaneTab('Results'));
+  };
+
+  const stopQuery = () => {
+    if (latestQuery && ['running', 'pending'].indexOf(latestQuery.state) >= 0) {
+      dispatch(postStopQuery(latestQuery));
+    }
+  };
+
+  const createTableAs = () => {
+    startQuery(true, CtasEnum.TABLE);
+    setShowCreateAsModal(false);
+    setCtas('');
+  };
+
+  const createViewAs = () => {
+    startQuery(true, CtasEnum.VIEW);
+    setShowCreateAsModal(false);
+    setCtas('');
+  };
+
+  const ctasChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCtas(event.target.value);
+  };
+
+  const queryPane = () => {
+    const hotkeys = getHotkeyConfig();
+    const { aceEditorHeight, southPaneHeight } =
+      getAceEditorAndSouthPaneHeights(height, northPercent, southPercent);
+    return (
+      <Split
+        expandToMin
+        className="queryPane"
+        sizes={[northPercent, southPercent]}
+        elementStyle={elementStyle}
+        minSize={200}
+        direction="vertical"
+        gutterSize={SQL_EDITOR_GUTTER_HEIGHT}
+        onDragStart={onResizeStart}
+        onDragEnd={onResizeEnd}
+      >
+        <div ref={northPaneRef} className="north-pane">
+          <AceEditorWrapper
+            actions={actions}
+            autocomplete={autocompleteEnabled}
+            onBlur={setQueryEditorSql}
+            onChange={onSqlChanged}
+            queryEditor={queryEditor}
+            sql={queryEditor.sql}
+            schemas={queryEditor.schemaOptions}
+            tables={queryEditor.tableOptions}
+            functionNames={queryEditor.functionNames}
+            extendedTables={tables}
+            height={`${aceEditorHeight}px`}
+            hotkeys={hotkeys}
+          />
+          {renderEditorBottomBar()}
+        </div>
+        <ConnectedSouthPane
+          editorQueries={editorQueries}
+          latestQueryId={latestQuery && latestQuery.id}
+          dataPreviewQueries={dataPreviewQueries}
+          actions={actions}
+          height={southPaneHeight}
+          displayLimit={displayLimit}
+          defaultQueryLimit={defaultQueryLimit}
+        />
+      </Split>
+    );
+  };
+
+  const renderDropdown = () => {
+    const qe = queryEditor;
+    const successful = latestQuery?.state === 'success';
+    const scheduleToolTip = successful
+      ? t('Schedule the query periodically')
+      : t('You must run the query successfully first');
+    return (
+      <Menu onClick={this.handleMenuClick} style={{ width: 176 }}>
+        <Menu.Item style={{ display: 'flex', justifyContent: 'space-between' }}>
```

Comment:
We should use emotion for styling both `Menu` and `Menu.Item` and avoid inline styles wherever possible